container/lxd/lxd.go

```diff
@@ -196,6 +212,41 @@ func (manager *containerManager) CreateContainer(
 	return
 }
 
+// getImageSources returns a list of LXD remote image sources based on the
+// configuration that was passed into the container manager.
+func (manager *containerManager) getImageSources() ([]lxdclient.Remote, error) {
+	imURL := manager.imageMetadataURL
+
+	// Unless the configuration explicitly requests the daily stream,
+	// an empty image metadata URL results in a search of the default sources.
+	if imURL == "" && manager.imageStream != "daily" {
+		logger.Debugf("checking default image metadata sources")
+		return lxdclient.DefaultImageSources, nil
+	}
+	// Otherwise only check the daily stream.
+	if imURL == "" {
+		return lxdclient.DefaultImageSources[1:], nil
+	}
+
+	imURL, err := imagemetadata.ImageMetadataURL(imURL, manager.imageStream)
+	if err != nil {
+		return nil, errors.Annotatef(err, "generating image metadata source")
+	}
+	// LXD requires HTTPS.
+	imURL = strings.Replace(imURL, "http:", "https:", 1)
+
+	remotes := []lxdclient.Remote{{
+		Name:          strings.Replace(imURL, "https://", "", 1),
```

Comment:
Didn't you just do this replacement 3 lines above here?

Reply:
That is just the name. I did that to mimic the Ubuntu sources.
```
Name = some.image.source
URL = https://some.image.source
```